src/test/java/com/rewedigital/composer/CompositionTest.java

```diff
@@ -0,0 +1,70 @@
+package com.rewedigital.composer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.spotify.apollo.Response;
+import com.spotify.apollo.test.ServiceHelper;
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+
+import okio.ByteString;
+
+public class CompositionTest {
+
+    @Rule
+    public final MockWebServer server = new MockWebServer();
+
+    @Rule
+    public ServiceHelper serviceHelper =
+        ServiceHelper.create(ComposerApplication.Initializer::init, ComposerApplication.COMPOSER)
+            .conf("composer.routing.local-routes", routesConfig(mockServerUrl()))
+            .conf("composer.html.include-tag", "include")
+            .conf("composer.html.content-tag", "content");
+
+    @Test
+    public void parsesMasterTemplateAndCombinesChildTemplate() throws Exception {
```

Comment:
When I read a test I like to see the connection between inputs and expected output. Here, the expected value `"4223"` appears out of nowhere... So, maybe an assertion that raises the level of abstraction by using a `Condition` is more appropriate: `asserThat(responseBody(composedFuture)).is(theCompositionOfMockServiceResponses())` or something like that?